falcon-regression/merlin-core/src/main/java/org/apache/falcon/regression/core/util/NativeInstanceUtil.java

```diff
@@ -0,0 +1,176 @@
+ * limitations under the License.
+ */
+
+package org.apache.falcon.regression.core.util;
+
+import org.apache.falcon.entity.v0.Entity;
+import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.entity.v0.Frequency;
+import org.apache.falcon.regression.core.helpers.ColoHelper;
+import org.apache.falcon.resource.InstancesResult;
+import org.apache.hadoop.security.authentication.client.AuthenticationException;
+import org.apache.log4j.Logger;
+import org.apache.oozie.client.CoordinatorAction;
+import org.testng.Assert;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+/**
+ * util functions related to instanceTest.
+ */
+public final class NativeInstanceUtil {
+
+    public static final int INSTANCES_CREATED_TIMEOUT = OSUtil.IS_WINDOWS ? 20 : 10;
+    private static final Logger LOGGER = Logger.getLogger(NativeInstanceUtil.class);
+
+    private NativeInstanceUtil() {
+        throw new AssertionError("Instantiating utility class...");
+    }
+
+    /**
+     * Waits till instance of specific entity will be created during timeout.
+     * Timeout is common for most of usual test cases.
+     *
+     * @param cluster     ColoHelper - colo on which API to be executed
+     * @param entity      definition of entity which describes job
+     * @param startTime   start time of instance
+     * @param endTime     end time of instance
+     */
+    public static void waitTillInstancesAreCreated(ColoHelper cluster, Entity entity, String startTime, String endTime)
+        throws InterruptedException, IOException, AuthenticationException,
+            URISyntaxException {
+        int sleep = INSTANCES_CREATED_TIMEOUT * 60 / 5;
+        waitTillInstancesAreCreated(cluster, entity, startTime, endTime, sleep);
+    }
+
+    /**
+     * Waits till instances of specific job will be created during specific time.
+     * Use this method directly in unusual test cases where timeouts are different from trivial.
+     * In other cases use waitTillInstancesAreCreated(ColoHelper,Entity,String,String)
+     *
+     * @param cluster     ColoHelper - colo on which API to be executed
+     * @param entity      definition of entity which describes job
+     * @param startTime   start time of instance
+     * @param endTime     end time of instance
+     * @param totalMinutesToWait     total time(in minutes) to wait for instance creation
+     */
+    public static void waitTillInstancesAreCreated(ColoHelper cluster, Entity entity, String startTime,
+        String endTime, int totalMinutesToWait)
+        throws URISyntaxException, AuthenticationException, InterruptedException, IOException {
+        String entityName = entity.getName();
+        EntityType type = entity.getEntityType();
+        String params = "?start=" + startTime;
+        params += (endTime.isEmpty() ? "" : "&end=" + endTime);
+
+        for (int sleepCount = 0; sleepCount < totalMinutesToWait; sleepCount++) {
```

Comment:
@pragya-mittal  IMO,60secs looks long. Because it may slow down our test execution. We can try with 5-10secs. Based on the future observation we can increase if it is required.